xonsh/jobs.py

```diff
@@ -251,9 +253,15 @@ def fg(args, stdin=None):
         act = tasks[0]  # take the last manipulated task by default
     elif len(args) == 1:
         try:
-            act = int(args[0])
-        except ValueError:
+            if args[0] == '+': # take the last manipulated task
+                 act = tasks[0]
+            elif args[0] == '-': # take the second to last manipulated task
+                act = tasks[1]
+            else:
+                act = int(args[0])
+        except (ValueError, IndexError):
             return '', 'Invalid job: {}\n'.format(args[0])
```

Comment:
@adqm that's right, fg should only ever take one or no arguments so it's fine to just grab the first one, and catching IndexError provides proper error handling in this case.